market/src/callbacks.rs

```diff
@@ -5,15 +5,22 @@ use crate::storage::*;
 #[near_bindgen]
 impl Market {
     #[private]
-    pub fn on_create_proposal_callback(&mut self) -> bool {
-        match env::promise_result(0) {
-            PromiseResult::Successful(_res) => {
-                //let proposal_id: u64 = near_sdk::serde_json::from_slice(&res).unwrap();
-                //self.market_options.push(proposal_id);
+    pub fn on_create_proposal_callback(&mut self) {
+        let market_options_len = self.market.options.len() as u64;
 
-                return true;
+        if env::promise_results_count() != market_options_len + 1 {
+            env::panic_str("ERR_CREATE_PROPOSALS_RESPONSES");
+        }
+
+        for n in 1..market_options_len + 1 {
+            match env::promise_result(n) {
+                PromiseResult::Successful(res) => {
+                    let proposal_id: u64 = near_sdk::serde_json::from_slice(&res).unwrap();
+                    self.proposals.push(proposal_id);
+                    self.published = true;
```

Comment:
This should be performed at the end of the loop.

Reply:
Fixed!